docs: clarify reschedule, migrate, and replacement terminology (#24929)
hashicorp/nomad: website/content/docs/job-specification/reschedule.mdx

task statuses become `failed`. The scheduler prefers to create a replacement
allocation on a node that was not used by a previous allocation.

> Rescheduling happens when the Nomad agent fails to setup the allocation or the
> tasks of an allocation fail more than their [`restart`][] block allows. When a
> node is drained, Nomad [migrates][] the allocations instead and the `reschedule`
> block does not apply. When a node is lost, Nomad [replaces][] the allocations
> instead and the `reschedule` block does not apply.

Reviewer: ```suggestion
Rescheduling happens when the Nomad agent fails to set up the allocation or the
tasks of an allocation fail more than their [`restart`][] block allows. When a
node is drained, Nomad [migrates][] the allocations instead and ignores the `reschedule`
block. When a node is lost, Nomad [replaces][] the allocations
instead and ignores the `reschedule` block.
```
